azurerm/resource_arm_mariadb_server.go

```diff
@@ -0,0 +1,381 @@
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"B_Gen5_1",
+								"B_Gen5_2",
+								"GP_Gen5_2",
+								"GP_Gen5_4",
+								"GP_Gen5_8",
+								"GP_Gen5_16",
+								"GP_Gen5_32",
+								"MO_Gen5_2",
+								"MO_Gen5_4",
+								"MO_Gen5_8",
+								"MO_Gen5_16",
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"capacity": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ValidateFunc: validate.IntInSlice([]int{
+								1,
+								2,
+								4,
+								8,
+								16,
+								32,
+							}),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(mariadb.Basic),
+								string(mariadb.GeneralPurpose),
+								string(mariadb.MemoryOptimized),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+
+						"family": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"administrator_login_password": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+
+			"storage_profile": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"storage_mb": {
+							Type:         schema.TypeInt,
+							Required:     true,
+							ValidateFunc: validate.IntBetweenAndDivisibleBy(5120, 4194304, 1024),
+						},
+
+						"backup_retention_days": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(7, 35),
+						},
+
+						"geo_redundant_backup": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(mariadb.Enabled),
+								string(mariadb.Disabled),
+							}, true),
+							DiffSuppressFunc: suppress.CaseDifference,
+						},
+					},
+				},
+			},
+
+			"ssl_enforcement": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(mariadb.SslEnforcementEnumDisabled),
+					string(mariadb.SslEnforcementEnumEnabled),
+				}, true),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+
+			tier, _ := diff.GetOk("sku.0.tier")
+			storageMB, _ := diff.GetOk("storage_profile.0.storage_mb")
+
+			if strings.ToLower(tier.(string)) == "basic" && storageMB.(int) > 1048576 {
```

Comment:
both of these fields are likely to be interpolated, as such it's likely the values won't be resolved at this point (due to a known bug in customizeDiff) and these values would be nil, so there'd be a crash here - as such I think we'd be better to put this logic in the Create method for the moment, unfortunately

Reply:
Thanks for the heads up about customizeDiff, moved the validation to the create method.